### DISCUSSION TITLE
How to have a position calculated if data is coming from backend for nodes

### DISCUSSION THREAD

**Discussion by kvedantmahajan** [xyflow/xyflow#2479] [Ideas]
All the examples are fine for rendering a static graph. For instance, the code below would be fine if I am constructing the position manually for each node. How would a dev be able to map the position to each of the node if data is a list of objects and unknown to client? 

```
{
  id: '1',
  type: 'input',
  data: { label: 'Node 1' },
  position: { x: 250, y: 5 }
}
```

Reference - [Node options](https://reactflow.dev/docs/api/nodes/node-options/)

Also, `position` as a node option is required as well and there is no default layout happening if it is skipped. 

Is there  a way out of this to do rendering from a real backend driven app or shall it be assumed that react-flow has a static rendering support for given data only?

**Comment by moklick** [comment]
For layouting you can use libs like:

* https://github.com/d3/d3-hierarchy
* https://github.com/dagrejs/dagre
* https://github.com/kieler/elkjs
